backends/tfhe-cuda-backend/cuda/src/integer/addition.cuh

```diff
@@ -0,0 +1,144 @@
+
+  *mem_ptr = new int_signed_overflowing_add_or_sub_memory<Torus>(
+      streams, gpu_indexes, gpu_count, params, num_blocks, op,
+      allocate_gpu_memory);
+}
+
+/*
+ * Addition - signed_operation = 1
+ * Subtraction - signed_operation = -1
+ */
+template <typename Torus, class params>
+__host__ void host_integer_signed_overflowing_add_or_sub_kb(
+    cudaStream_t *streams, uint32_t *gpu_indexes, uint32_t gpu_count,
+    Torus *lhs, Torus *rhs, Torus *overflowed, SIGNED_OPERATION op, void **bsks,
+    uint64_t **ksks,
+    int_signed_overflowing_add_or_sub_memory<uint64_t> *mem_ptr,
+    uint32_t num_blocks) {
+
+  auto radix_params = mem_ptr->params;
+
+  uint32_t big_lwe_dimension = radix_params.big_lwe_dimension;
+  uint32_t big_lwe_size = big_lwe_dimension + 1;
+  uint32_t big_lwe_size_bytes = big_lwe_size * sizeof(Torus);
+
+  assert(radix_params.message_modulus >= 4 && radix_params.carry_modulus >= 4);
+
+  auto result = mem_ptr->result;
+  auto input_carries = mem_ptr->input_carries;
+  auto output_carry = mem_ptr->output_carry;
+  auto last_block_inner_propagation = mem_ptr->last_block_inner_propagation;
+
+  cuda_memcpy_async_gpu_to_gpu(result, lhs, num_blocks * big_lwe_size_bytes,
+                               streams[0], 0);
+
+  // phase 1
+  if (op == SIGNED_OPERATION::ADDITION) {
+    host_addition(streams[0], gpu_indexes[0], result, lhs, rhs,
+                  big_lwe_dimension, num_blocks);
+  } else {
+    host_subtraction(streams[0], gpu_indexes[0], result, lhs, rhs,
+                     big_lwe_dimension, num_blocks);
+  }
+
+  // phase 2
+  for (uint j = 0; j < gpu_count; j++) {
+    cuda_synchronize_stream(streams[j], gpu_indexes[j]);
+  }
+
+#pragma omp parallel sections
+  {
+    // generate input_carries and output_carry
+#pragma omp section
+    {
+      host_propagate_single_carry(
+          mem_ptr->sub_streams_1, gpu_indexes, gpu_count, result, output_carry,
+          input_carries, mem_ptr->scp_mem, bsks, ksks, num_blocks);
+    }
+
+    // generate generate_last_block_inner_propagation
+#pragma omp section
+    {
+      host_generate_last_block_inner_propagation(
+          mem_ptr->sub_streams_2, gpu_indexes, gpu_count,
+          last_block_inner_propagation, &lhs[(num_blocks - 1) * big_lwe_size],
+          &rhs[(num_blocks - 1) * big_lwe_size], mem_ptr->las_block_prop_mem,
+          bsks, ksks);
+    }
+  }
+
+  for (uint j = 0; j < gpu_count; j++) {
+    cuda_synchronize_stream(mem_ptr->sub_streams_1[j], gpu_indexes[j]);
+    cuda_synchronize_stream(mem_ptr->sub_streams_2[j], gpu_indexes[j]);
+  }
+
+  // phase 3
+  auto input_carry = &input_carries[(num_blocks - 1) * big_lwe_size];
+
+  host_resolve_signed_overflow(
+      streams, gpu_indexes, gpu_count, overflowed, last_block_inner_propagation,
+      input_carry, output_carry, mem_ptr->resolve_overflow_mem, bsks, ksks);
+
+  cuda_memcpy_async_gpu_to_gpu(lhs, result, num_blocks * big_lwe_size_bytes,
+                               streams[0], 0);
```

Comment:
You should replace:
```
streams[0], 0);
```
by: 
```
streams[0], gpu_indexes[0]);
```
With a single GU the user has the possibility to pass a GPU index different from 0.